internal/server/flag.go

```diff
@@ -65,6 +67,14 @@ func (s *Server) ListFlags(ctx context.Context, r *flipt.ListFlagRequest) (*flip
 
 	results, err := s.store.ListFlags(ctx, r.NamespaceKey, opts...)
 	if err != nil {
+		var jerr *json.SyntaxError
+
+		// Handle special case for failure to decode page token.
+		if errors.As(err, &jerr) {
+			s.logger.Debug("Failed to decode page token", zap.Error(err))
+			return nil, ferrors.ErrInvalidf("pageToken is not valid: %q", r.PageToken)
+		}
```

Comment:
Apologies, as I know this is going to be a bit of a lift and shift.:
However, could we move this error handing into the stores themselves.

One of the ideas around using page token (instead of limit / offset) was that it would allow the storage layer to decide how to handle pagination. i.e. what goes in the page token payload.
Whether or not these page tokens are json ended is really down to what the storage layer decides when it encodes a next page token. The server layer just trusts the storage layer to make that decision. Different storage layers might choose to encode these differently (e.g. one might choose protobuf (probably wont, but that is the intention of responsibility)).